### PR TITLE
Fetch message for quotefast action via Msg object

### PR DESCRIPTION
**Pull request by live627**
### Motivation

I have a modification (Topic Descriptions) that needs access to this, so we'll use `Msg::load()` which eventually calls `integrate_query_message`

### Idea

`$query_customizations` could become its own object.

### CODE REVIEW

**Comment by jdarwood007** [comment]
Haven't run the code but looks good.

I'm in support of more objects for everything.  It makes things much easier to develop.  But we have to contend with how it would support customizations.  Magic methods maybe can help there.

**Comment by DiegoAndresCortes** [comment]
I like it, same thing we got for messageindex, the more places the better imo

**Comment by tyrsson** [comment]
From a customization perspective. If you model the instances that need to support customization as Observable then mods can be notified via their own registered listeners.

File: Sources/Actions/QuoteFast.php [hunk 1]
```diff
@@ -47,52 +48,68 @@ class QuoteFast implements ActionInterface
...
-		Db::$db->free_result($request);
+		];
 
-		Utils::$context['sub_template'] = 'quotefast';
+		$bq = User::$me->mod_cache['bq'];
 
-		if (!empty($row)) {
-			$can_view_post = $row['approved'] || ($row['id_member'] != 0 && $row['id_member'] == User::$me->id) || User::$me->allowedTo('approve_posts', $row['id_board']);
+		if (isset($_REQUEST['modify']) || $bq != '1=1') {
+			$query_customizations['where'][] = 't.locked = {int:not_locked}' . ($bq == '0=1' ? '' : 'm.' . $bq);
```
Comment (Sesquipedalian) [inline Sources/Actions/QuoteFast.php]:
```suggestion
			$query_customizations['where'][] = 't.locked = {int:not_locked}' . ($bq == '0=1' ? '' : ' OR m.' . $bq);
```

Is this what you meant to write?

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**